paddle/fluid/operators/mkldnn/softmax_mkldnn_op.cc

```diff
@@ -154,21 +211,15 @@ class SoftmaxMKLDNNKernel : public paddle::framework::OpKernel<T> {
     const std::string key =
         platform::MKLDNNHandler::GetHash(softmax_tz, ctx.op().Output("Out"));
 
-    SoftmaxMKLDNNHandler handler(dev_ctx, mkldnn_engine, key);
-    // Currently only NC data format is supported
-    auto softmax_md = MKLDNNMemDesc(
-        {softmax_tz}, platform::MKLDNNGetDataType<T>(), memory::format::nc);
-    // Normalization is made after innermost dimension eg. C out of NC
-    auto softmax_desc = softmax_forward::desc(prop_kind::forward_scoring,
-                                              softmax_md, 1 /*dim: C*/);
-
-    auto softmax_pd =
-        handler.AcquireSoftmaxPrimitiveDescriptor(softmax_desc, mkldnn_engine);
+    SoftmaxMKLDNNHandler handler(
+        softmax_tz, input->type(), platform::MKLDNNGetDataType<T>(),
+        mkldnn::memory::format::nc, dev_ctx, mkldnn_engine, key);
```

Comment:
I think we should consider hiding key creation inside AcquireAPI too.

Reply:
That is actually very good idea to be implemented. I have added it to the list in relevant refactoring issue #18531 so it can be done later on.